pump/server.go

```diff
@@ -399,14 +399,16 @@ func (s *Server) Start() error {
 
 	log.Info("register success", zap.String("NodeID", s.node.NodeStatus().NodeID))
 
-	// notify all cisterns
-	ctx, _ := context.WithTimeout(s.ctx, notifyDrainerTimeout)
-	if err := s.node.Notify(ctx); err != nil {
-		// if fail, refresh this node's state to paused
-		if err := s.registerNode(context.Background(), node.Paused, 0); err != nil {
-			log.Error("unregister pump while pump fails to notify drainer", zap.Error(err))
+	// notify all cisterns when this pump node is a newly registered one
+	if s.node.NodeStatus().State != node.Paused {
+		ctx, _ := context.WithTimeout(s.ctx, notifyDrainerTimeout)
+		if err := s.node.Notify(ctx); err != nil {
+			// if fail, refresh this node's state to paused
+			if err := s.registerNode(context.Background(), node.Paused, 0); err != nil {
```

Comment:
start a new pump -> notify failed -> set to be Paused -> start again -> skip notify  ?

Reply:
Exactly. I've changed `node.Paused` to `node.Offline`.